stix_shifter_modules/darktrace/stix_transmission/connector.py

```diff
@@ -0,0 +1,197 @@
+                results.append(element_dict)
+
+        return results
+
+    @staticmethod
+    def check_empty_data(response):
+        """
+        Adding protocol, file name value and remove empty data
+        params: res_dict(dict) response dictionary
+        return: res_dict(dict)
+        """
+
+        file_fields = ["filename", "sha1_file_hash", "sha1", "md5_file_hash", "md5", "sha256_file_hash", "sha256",
+                       "mime", "file_mime_type", "mime_type", "dcc_mime_type", "total_bytes", "file_msg", "read_size",
+                       "write_size", "dcc_file_size"]
+
+        for index in range(len(response['data'])):
+            for key in response['data'][index]:
+                if response['data'][index][key].get("filename") is None:
+                    if response['data'][index][key].keys() & file_fields:
+                        response['data'][index][key]["filename"] = "null"
+
+                empty_keys = [k for k, v in response['data'][index][key].items() if not v]
+                for empty_key in empty_keys:
+                    del response['data'][index][key][empty_key]
+        return response
+
+    def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: dict
+        """
+        return_obj = {}
+        response_dict = {}
+        try:
+            response = self.api_client.ping_box()
+            response_code = response.response.status_code
+            response_dict = json.loads(response.response.text)
+
+            if response_code == 200:  # and response_dict['status'] == 'SUCCESS':
+                return_obj['success'] = True
+            elif response_code == 400:
+                raise InvalidAuthenticationException
+
+            if response_dict.get('error'):
+                raise InvalidArguments(response_dict['error'])
+
+        except InvalidArguments as ex:
+            response_dict['type'] = 1002
+            response_dict['message'] = str(ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except InvalidAuthenticationException:
+            response_dict['code'] = 1001
+            response_dict['message'] = "Invalid Authentication"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except ConnectionError:
+            response_dict['code'] = 1003
+            response_dict['message'] = "Invalid Host/Port"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except Exception as ex:
+            response_dict['type'] = ex.__class__.__name__
+            response_dict['message'] = ex
+            self.logger.error('error while pinging: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    def delete_query_connection(self, search_id):
```

Comment:
you can remove `delete_query_connection()`

Reply:
Removed delete_query_connection() function as suggested.